app/models/king.rb

```diff
@@ -1,5 +1,28 @@
 class King < Piece
   scope :piece, -> {where(type: 'King')}
 
+  def valid_move?(piece_destination)
+    piece[0] = x_current
+    piece[1] = y_current
+    piece_destination[0] = x_destination
+    piece_destination[1] = y_destination
 
-end
+    # If piece_destination is not within the board -- not allowed
+    if x_destination < 1 || x_destination > 8 || y_destination < 1 || y_destination > 8
```

Comment:
we could make this a separate method and call it sth like `if outside_board(x_dest, y_dest)`..